File: python/multiple_queries.py
from pprint import pprint
import requests
import os

OS_API_KEY = os.getenv("OS_API_KEY")
if not OS_API_KEY:
    raise ValueError("The OS_API_KEY environment variable is not set")


def main():
    # Prepare authentication using the Authorization header
    headers = {
        "Authorization": OS_API_KEY,
    }

    # Prepare two queries for one request
    query = {
        "queries": {
            "query-A": {
                "schema": "Person",
                "properties": {
                    "name": ["Arkadiii Romanovich Rotenberg", "Ротенберг Аркадий"],
                    "birthDate": ["1951"],
                },
            },
            "query-B": {
                "schema": "Company",
                "properties": {"name": ["Stroygazmontazh"], "jurisdiction": ["Russia"]},
            },
        }
    }

    # Make the request
    response = requests.post(
        "https://api.opensanctions.org/match/default", headers=headers, json=query
    )

    # Check for HTTP errors
    response.raise_for_status()

    for query_id in query["queries"].keys():
        print(f"\nResults for query {query_id}:")
        # Get the ID, name, match, score, and features for each result
        results = []
        for match in response.json()["responses"][query_id]["results"]:
            results.append(
                {
                    "id": match["id"],
                    "name": match["properties"]["name"],
                    "match": match["match"],
                }
            )

        # print it out with nice formatting
        pprint(results, sort_dicts=False)


if __name__ == "__main__":
    main()

File: node-js/multiple_queries.js
const OS_API_KEY = process.env.OS_API_KEY;
if (!OS_API_KEY) {
  console.log('Please set the OS_API_KEY environment variable');
  process.exit(1);
}

async function match() {
  const headers = {
    // Prepare authentication using the Authorization header
    "Authorization": OS_API_KEY,

    // We're going to encode the request body as JSON.
    'Content-Type': 'application/json',
  };

  // Prepare a query to match on schema and the name property
  const requestBody = {
    "queries": {
      "query-A": {
        "schema": "Person",
        "properties": {
          "name": ["Arkadiii Romanovich Rotenberg", "Ротенберг Аркадий"],
          "birthDate": ["1951"],
        },
      },
      "query-B": {
        "schema": "Company",
        "properties": { "name": ["Stroygazmontazh"], "jurisdiction": ["Russia"] },
      },
    }
  }

  // Construct the request URL
  const url = new URL('https://api.opensanctions.org/match/default');
  url.searchParams.set('algorithm', 'regression-v1');

  // Make the request
  const response = await fetch(
    url,
    {
      'method': 'POST',
      'headers': headers,
      'body': JSON.stringify(requestBody),
    });

  // Check for errors
  if (!response.ok) {
    console.error(await response.text());
    throw new Error('HTTP error\n' + response.status + ' ' + response.statusText);
  }

  const responseBody = await response.json();

  // Get the results for each query
  for (const [queryId, query] of Object.entries(requestBody.queries)) {
    console.log(`\nResults for query ${queryId}:`)

    // We look up the query results by ID in the responses object
    const results = responseBody.responses[queryId].results.map((result) => {
      return {
        "id": result.id,
        "name": result.properties.name,
        "match": result.match,
      }
    });

    // We're just JSON encoding the response data here for pretty printing
    console.log(JSON.stringify(results, null, 2));
  };
}

match()
  .then()
  .catch((error) => console.log("Error:" + error.message));
